lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb

```diff
@@ -840,31 +842,39 @@ def cmd_migrate_help
   #   platforms a path for the unix domain socket used for IPC.
   # @return [void]
   def cmd_migrate(*args)
-    if args.length == 0 || args.include?('-h')
+    if args.length == 0 || args.any? { |arg| %w(-h --pid --name).include? arg }
       cmd_migrate_help
       return true
     end
 
-    pid = args[0].to_i
-    if pid == 0
-      print_error('A process ID must be specified, not a process name')
-      return
-    end
-
+    pid = nil
     writable_dir = nil
     opts = {
       timeout: nil
     }
 
-    @@transport_opts.parse(args) do |opt, idx, val|
```

Comment:
Thanks Jon. I had a PR ready for this options parsing issue, but fur some
reason hadn't pushed it yet.

Did you know about "run migrate -n process.exe" ?

Cheers!